### ISSUE TITLE
idea for API

### ISSUE THREAD

**Issue by jonschlinkert** [doowb/macro-store#1]
I think we should change this API to be stupid simple and not expose methods. Here's how it would work:

``` js
var argv = require('macro-store')(process.argv.slice(2));
var args = require('yargs-parser')(argv);
```

macro-store would determine if a macro is being stored, or if the first argument on `process.argv` is the key of a stored macro. If it is, then macro store would return the stored arguments array.

This way macros are used transparently before any processing by the application, and you turn macro-store into a library that can be used with every CLI in node.js.

**Comment by jonschlinkert** [comment]
I think there is room for creativity in regards to how macros are stored. Meaning that it doesn't need to be just an array. We should start by making it simple though, so we can use it and make changes based on need.

**Comment by doowb** [comment]
Check out the [refactor](https://github.com/doowb/macro-store/tree/refactor) branch. I think this is what you're getting at.

When setting and deleting macros, the function returns an empty array as the results. We could change that if we want to somehow indicate back to the implementing code what happened.

Running `node example.js` with some commands:

![image](https://cloud.githubusercontent.com/assets/995160/17566727/99fff650-5f0a-11e6-9d3d-a4f4132666bb.png)

**Comment by jonschlinkert** [comment]
yeah that's exactly what I was thinking!

**Comment by jonschlinkert** [comment]
maybe the returned value should be something like: 

``` js
{ macro: { foo: ['a', 'b', 'c'] }}
```

so implementors can log out what was saved

**Comment by jonschlinkert** [comment]
I wonder if we should allow 

``` sh
$ app --macro foo bar baz
```

Or

``` sh
$ app foo --macro bar baz
```

It seems like the first one makes more sense. I know I suggested the second, but maybe I was wrong

**Comment by jonschlinkert** [comment]
actually maybe you should use yargs-parser or something, then you can explain in the docs something like "If yargs-parser returns an object with a `macro` property, the value of `macro` is used as the key, and the rest of the original arguments array is saved as the value"

edit: then you could also return the original argv array as well as the parsed object. The advantage of this is that implementors wouldn't need to use yargs-parser or minimist. You could allow implementors to pass a `parser` on the options though

**Comment by doowb** [comment]
yeah, i like the parser idea. i was thinking that might be a better way to go when you mentioned changing where the `--macro` option is in the array.

**Comment by doowb** [comment]
Updated with what we discussed.

Screenshots for docs:

### Setting

![image](https://cloud.githubusercontent.com/assets/995160/17576116/6b11e06e-5f3d-11e6-8019-6ad574c5fc16.png)

### Getting multiple

![image](https://cloud.githubusercontent.com/assets/995160/17576131/8940e558-5f3d-11e6-8a57-20bf52328fe0.png)

### Deleting one

![image](https://cloud.githubusercontent.com/assets/995160/17576151/aa2e14ac-5f3d-11e6-9a1f-2622264f1be2.png)

### Deleting all

![image](https://cloud.githubusercontent.com/assets/995160/17576156/bfdec8fa-5f3d-11e6-8947-2f4570b695ee.png)

**Comment by jonschlinkert** [comment]
nice!

**Comment by doowb** [comment]
Merged and published as `0.2.0`.
